src/components/NavBar/NavBar.vue

```diff
@@ -50,40 +52,81 @@
 </template>
 
 <script>
+import axios from "../../axios/axios.v1";
 import { logout } from "../../hooks/useCredential.js";
 export default {
   name: "NavBar",
   data() {
     return {
-      stocks: [
-        { symbol: "AAPL", name: "Apple Inc." },
-        { symbol: "TSLA", name: "Tesla Inc." },
-        { symbol: "FSR", name: "Fisker Inc." },
-        { symbol: "SPI", name: "Spi Energy Co Ltd" },
-      ],
+      stocks: [],
       isLoading: false,
+      searchModel: null,
+      search: null,
+      searchQueue: [],
     };
   },
+  computed: {
+    fields() {
+      if (!this.searchModel) return [];
+      return Object.keys(this.searchModel).map((key) => {
+        return {
+          key,
+          value: this.searchModel[key] || "n/a",
+        };
+      });
+    },
+    items() {
+      return this.stocks.map((stock) => {
+        const Description = stock["1. symbol"] + ": " + stock["2. name"];
+        return Object.assign({}, stock, { Description });
+      });
+    },
+  },
   methods: {
     submit() {
       logout(this.$store, this.$router);
     },
-    redirect(symbol){
+    redirect(symbol) {
       const regex = /^\S+$/;
-      if(regex.test(symbol)){
-        window.open("https://finance.yahoo.com/quote/"+symbol);
+      if (regex.test(symbol)) {
+        window.open("https://finance.yahoo.com/quote/" + symbol);
       }
-      return true
-    }
+      return true;
+    },
   },
-  computed: {
-    items() {
-      return null;
-    }
+  watch: {
+    async search(val) {
+      // Checks if the user has typed anything in the last 2 seconds if not make a request to backend
+      this.searchQueue.push(val);
+      await new Promise((_) => setTimeout(_, 2000));
+      this.searchQueue.pop();
+      if (this.searchQueue.length == 0) {
```

Comment:
got this error when I clicked outside the search bar and I also think it's better to use `===` instead of `==` to prevent type coercion 
suggestion: use `?.` operator: `this?.searchQueue.length === 0`
![image](https://user-images.githubusercontent.com/30036024/151079418-0ece2fdf-6097-40e1-97e5-a90a8c57f2a0.png)

Reply:
searchQueue is initialized as an empty list so we know it will always work with length

Reply:
I'll add it because it's extra precaution so it's not a bad idea